sympy/core/containers.py

```diff
@@ -304,6 +304,13 @@ def __lt__(self, other):
     def _sorted_args(self):
         return tuple(sorted(self.args, key=default_sort_key))
 
+    def __eq__(self, other):
+        if isinstance(other, dict):
+            return self == Dict(other)
+        return super().__eq__(other)
```

Comment:
Would it not be better to do `return self._dict == other`?

Reply:
That would currently fail, since `Dict` uses `sympify` (in `Tuple`), it would try to match e.g. `Symbol('x')` with `'x'`:
`Dict({'x':1}) == {'x':1}` would return `False`.

This requires a larger `Dict` overhaul that I can hopefully solve in https://github.com/sympy/sympy/pull/22580

Reply:
Tuple should be fixed to use `_sympify`.

Reply:
I agree, but it requires a large cleanup of parts in the code that rely on `sympify` being used in `Tuple`